crates/ark/src/variables/variable.rs

```diff
@@ -918,132 +931,216 @@ impl PositronVariable {
+                })
+            },
+            LGLSXP | RAWSXP | STRSXP | INTSXP | REALSXP | CPLXSXP => {
+                if r_is_matrix(object.sexp) {
+                    Ok(EnvironmentVariableNode::Matrixcolumn {
                         object,
-                        index: n_row * index + row_index,
-                    }
-                },
-            }
+                        index: parse_index(access_key)?,
+                    })
+                } else {
+                    Ok(EnvironmentVariableNode::AtomicVectorElement {
+                        object,
+                        index: parse_index(access_key)?,
+                    })
+                }
+            },
+            _ => Err(harp::Error::Anyhow(anyhow!(
+                "Unexpected child at {access_key}"
+            ))),
+        }
+    }
+
+    fn get_child_node_at(
+        node: EnvironmentVariableNode,
+        path_elt: &String,
+    ) -> harp::Result<EnvironmentVariableNode> {
+        match node {
+            EnvironmentVariableNode::Concrete { object } => {
+                Self::get_concrete_child_node(object, path_elt)
+            },
+
+            EnvironmentVariableNode::R6Node { object, name } => {
+                match name.as_str() {
+                    "<private>" => {
+                        let env = Environment::new(object);
+                        let enclos = Environment::new(RObject::view(env.find(".__enclos_env__")?));
+                        let private = Environment::new(RObject::view(enclos.find("private")?));
+
+                        // TODO: it seems unlikely that private would host active bindings
+                        //       so find() is fine, we can assume this is concrete
+                        Ok(EnvironmentVariableNode::Concrete {
+                            object: RObject::view(private.find(path_elt)?),
+                        })
+                    },
+
+                    _ => {
+                        // Technically we'd also implement this for `<methods>`, but because `methods`
+                        // are all functions which always `have_children=false` we don't need to.
+                        return Err(harp::Error::Anyhow(anyhow!(
+                            "You can only get children from <private>, got {path_elt}"
+                        )));
+                    },
+                }
+            },
+
+            EnvironmentVariableNode::AtomicVectorElement { .. } => {
+                return Err(harp::Error::Anyhow(anyhow!(
+                    "Can't subset an atomic vector even further, got {path_elt}"
+                )));
+            },
+
+            EnvironmentVariableNode::Matrixcolumn { object, index } => unsafe {
+                let dim = IntegerVector::new(Rf_getAttrib(*object, R_DimSymbol))?;
+                let n_row = dim.get_unchecked(0).unwrap() as isize;
+
+                // TODO: use ? here, but this does not return a crate::error::Error, so
```

Comment:
yep it's fine to return an `Error::Anyhow` variant.

Reply:
Fixed in https://github.com/posit-dev/ark/pull/561/commits/c57868a9258e8c9bb697cc718d3a8bb342930585